proto/dwtools/abase/l0.test/Buffer.test.s

```diff
@@ -10014,39 +10014,160 @@ function bufferResize_( test )
 
 //
 
+function bufferBytesGet( test )
+{
+
+  test.case = 'from string';
+  var src = 'string';
+  var got = _.bufferBytesGet( src );
+  var expected = new U8x([ 115, 116, 114, 105, 110, 103 ]);
+  test.identical( got, expected );
+
+  /* BufferRaw */
+
+  test.case = 'from BufferRaw';
+  var src = new BufferRaw( 5 );
+  var got = _.bufferBytesGet( src );
+  var expected = new U8x( [ 0, 0, 0, 0, 0, ] );
+  test.identical( got, expected );
+
+  test.case = 'from BufferRawShared';
+  var src = new BufferRawShared( 5 );
+  var got = _.bufferBytesGet( src );
+  var expected = new U8x( [ 0, 0, 0, 0, 0, ] );
+  test.identical( got, expected );
+
+  /* BufferNode */
+
+  test.case = 'from BufferNode (array)';
+  var src = BufferNode.from( [ 2, 4 ] );
+  var got = _.bufferBytesGet( src );
+  var expected = new U8x( [ 2, 4 ] );
+  test.identical( got, expected );
+
+
+  test.case = 'from BufferNode (string)';
+  var src = BufferNode.alloc( 5, 'a' );
+  var got = _.bufferBytesGet( src );
+  var expected = new U8x( [ 97, 97, 97, 97, 97 ] );
+  test.identical( got, expected );
+
+  /* BufferTyped */
+
+  test.case = 'from typed array I16x';
+  var src = new I16x( [ 1, 2, 10 ] );
+  var got = _.bufferBytesGet( src );
+  var expected = new U8x( [ 1, 0, 2, 0, 10, 0 ] );
+  test.identical( got, expected );
+
+  test.case = 'from typed array F32x';
+  var src = new F32x( [ 1, 2 ] );
+  var got = _.bufferBytesGet( src );
+  var expected = new U8x( [ 0, 0, 128, 63, 0, 0, 0, 64 ] );
+  test.identical( got, expected );
+
+  /* BufferView */
+
+  test.case = 'from BufferView'
+  var src = new BufferView( new BufferRaw( 5 ) );
+  var got = _.bufferBytesGet( src );
+  var expected = new U8x( [ 0, 0, 0, 0 ] );
+  test.identical( got, expected );
+
+  /**/
+
+  if( !Config.debug )
+  return;
+
+  test.case = 'without argument';
+  test.shouldThrowErrorSync( () => _.bufferBytesGet() );
+
+  test.case = 'extra arguments';
+  test.shouldThrowErrorSync( () => _.bufferBytesGet( new BufferRaw( 5 ) ), 'extra' );
+
+  test.case = 'wrong buffer type';
+  test.shouldThrowErrorSync( () => _.bufferBytesGet( [ 1, 2 ] ) );
+  test.shouldThrowErrorSync( () => _.bufferBytesGet( 1 ) );
+}
+
+//
+
 function bufferRetype( test )
 {
 
-  test.case = 'converts and returns the new type of I16x';
+  /* BufferRaw */
+
+  test.case = 'from BufferRaw, return the new type of U8x';
+  var src = new BufferRaw( 5 );
+  var got = _.bufferRetype( src, U8x );
+  var expected = new U8x( [ 0, 0, 0, 0, 0, ] );
+  test.identical( got, expected );
+
+  test.case = 'from BufferRawShared, return the new type of U8x';
+  var src = new BufferRawShared( 5 );
+  var got = _.bufferBytesGet( src,  U8x );
+  var expected = new U8x( [ 0, 0, 0, 0, 0, ] );
+  test.identical( got, expected );
+
+  /* BufferNode */
+
+  test.case = 'from BufferNode, return the new type of U16x';
```

Comment:
Please, insert group `BufferNode` in `if` block

```js
if( Config.interpreter === 'njs' )
{
  // code
}
```
`BufferNode` exists  in `NodeJs`.
It prevents exception in another interpreters.